lib/greeter.rb

```diff
@@ -2,4 +2,8 @@ class Greeter
   def greet
     'Hello!'
   end
+  def greet_by_name name
+    return self.greet if name == '' or name.class != String
```

Comment:
Hey, it would probably be good to add a test for the class check to be sure this keeps working.

Reply:
Added a couple tests for this case. Thanks.